src/App.tsx

```diff
@@ -12,22 +12,37 @@ import {
 import appTheme from './Theme';
 import lightTheme from './LightTheme';
 import OpenSimAppBar from './components/Nav/OpenSimAppBar';
-import React from 'react';
+import React, { useEffect } from 'react';
 
 function App() { 
 
   const [dark, setDark] = React.useState(true);
-  window.addEventListener("keyup", (event) => {
-    if (event.code==='KeyD'){ // P for print screen
-      setDark(!dark);
-    }
-  });
+
+  useEffect(() => {
+    // Event that switches between dark and light mode when the letter D is pressed.
+    const handleKeyDUp = (event: KeyboardEvent) => {
+      if (event.code === "KeyD") {
+        setDark((dark) => !dark);
+      }
+    };
+
+    // Register the event to "keyup" when the element is mount
+    window.addEventListener("keyup", handleKeyDUp);
+
+    return () => {
+      // Unregister the event when the element is unmount.
+      window.removeEventListener("keyup", handleKeyDUp);
+    };
+  }, []);
+
+  console.log("Rendering")
+
```

Comment:
Should we remove this debugging line, or is there console log that only triggers in dev/debug mode?

Reply:
Sure, we can remove it. I only added it at the beginning to check if the bug was caused because of renders growing exponentially.